src/components/Wrappers/withUserRoles.js

```diff
@@ -69,7 +80,11 @@ const withUserRoles = (WrappedComponent) => (props) => {
           userId,
           roleIds: roleIds[tenantIdKey],
         }
-      }).catch(sendErrorCallout);
+      })
+        .then(async () => {
+          await queryClient.invalidateQueries({ queryKey: [namespace, userId, tenantIdKey] });
+        })
+        .catch(sendErrorCallout);
```

Comment:
Sonar is grumpy about how deeply nested this is. Let's convert it all to async/await instead of mixing that with promise chaining.